resources/tests/test_resource_api.py

```diff
@@ -136,11 +137,51 @@ def test_non_public_resource_visibility(api_client, resource_in_unit, user):
     response = api_client.get(url)
     assert response.status_code == 200
     assert response.data['count'] == 1
-
     url = reverse('resource-detail', kwargs={'pk': resource_in_unit.pk})
     response = api_client.get(url)
     assert response.status_code == 200
 
+    # Authenticated as unit admin
+    user.is_general_admin = False
+    user.save()
+    user.unit_authorizations.create(
+        authorized=staff_user,
+        level=UnitAuthorizationLevel.admin,
+        subject=resource_in_unit.unit
+    )
+    user.save()
+    url = reverse('resource-list')
+    response = api_client.get(url)
+    assert response.status_code == 200
+    assert response.data['count'] == 1
+    assert Unit.objects.managed_by(user).values_list('id', flat=True)[0] == response.data['results'][0]['unit']
+
+    # Authenticated as unit group admin
+    user.unit_authorizations.all().delete()
+    unit_group = UnitGroup.objects.create(name='foo')
+    unit_group.members.add(resource_in_unit.unit)
+    user.unit_group_authorizations.create(
+        authorized=staff_user,
+        level=UnitGroupAuthorizationLevel.admin,
+        subject=unit_group
+    )
+    user.save()
+    url = reverse('resource-list')
+    response = api_client.get(url)
+    assert response.status_code == 200
+    assert response.data['count'] == 1
+    assert Unit.objects.managed_by(user).values_list('id', flat=True)[0] == response.data['results'][0]['unit']
+
+    # Authenticated as unit manager
+    user.unit_authorizations.level = UnitAuthorizationLevel.manager
+    user.unit_authorizations.subject = resource_in_unit.unit
+    user.save()
```

Comment:
This is still incorrect and relying to the UnitGroup Administrator status set on line 163. Maybe you could move this as the first one of these three new tests and then in the second test just change the level of the UnitAuthorization object from manager to admin.

Otherwise these tests look fine now.